pandas/parser.pyx

```diff
@@ -548,6 +555,29 @@ cdef class TextReader:
     def set_error_bad_lines(self, int status):
         self.parser.error_bad_lines = status
 
+    def _set_quoting(self, quote_char, quoting):
+        if not isinstance(quoting, int):
+            raise TypeError('"quoting" must be an integer')
+
+        if not 0 <= quoting <= 3:
```

Comment:
you should test that instead it is a valid value (e.g. it has to match one of the constants)